src/Symfony/Component/Console/Question/Question.php

```diff
@@ -160,6 +169,8 @@ public function setAutocompleterValues(?iterable $values): static
 
     /**
      * Gets the callback function used for the autocompleter.
+     *
+     * @return (callable(mixed):iterable)|null
```

Comment:
```suggestion
     * @return (callable(string):string[])|null
```

The autocomplete callback is expected to return an array of strings. We don't support returning an iterator (we use array functions with that value for instance) and values must be strings.
the current value from which to autocomplete is also guaranteed to be a string.

btw, the same type is useful for `\Symfony\Component\Console\Helper\QuestionHelper::autocomplete` (that is consuming the value returned by this getter), so you could update it as well.